onnxruntime/python/tools/quantization/onnx_quantizer.py

```diff
@@ -0,0 +1,925 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import onnx
+import onnx.numpy_helper
+import struct
+from pathlib import Path
+
+import numpy as np
+
+from onnx import onnx_pb as onnx_proto
+from onnx import shape_inference
+from onnxruntime import SessionOptions, InferenceSession, GraphOptimizationLevel
+
+from .quant_utils import QuantizationMode, QuantizedValueType, QuantizedInitializer, QuantizedValue, quantization_modes
+from .quant_utils import _find_by_name, _get_elem_index, _get_mul_node, _generate_identified_filename, _attribute_to_kwarg
+from .quant_utils import QuantType, onnx_domain, __producer__, __version__
+
+from .registry import CreateOpQuantizer, CreateDefaultOpQuantizer
+
+from .onnx_model import ONNXModel
+
+
+def quantize_data(data, quantize_range, qType):
+    '''
+        :parameter data: data to quantize
+        :parameter quantize_range: list of data to weight pack.
+        :parameter qType: data type to quantize to. Supported types UINT8 and INT8
+        :return: minimum, maximum, zero point, scale, and quantized weights
+        To pack weights, we compute a linear transformation
+            - when data type == uint8 mode, from [rmin, rmax] -> [0, 2^{b-1}] and
+            - when data type == int8, from [-m , m] -> [-(2^{b-1}-1), 2^{b-1}-1] where
+                m = max(abs(rmin), abs(rmax))
+        and add necessary intermediate nodes to trasnform quantized weight to full weight using the equation
+        r = S(q-z), where
+            r: real original value
+            q: quantized value
+            S: scale
+            z: zero point
+    '''
+    rmin = min(min(data), 0)
```

Comment:
what's usage of 0 in rmin and rmax?   To clip, it is like max(min(data), 0)

Reply:
need to make sure rmax>=0, and rmin <=0